smalltalksrc/Slang-Tests/SlangBasicTranslationTest.class.st

```diff
@@ -1321,6 +1359,96 @@ SlangBasicTranslationTest >> testSendTimes [
 	self assert: translation equals: '(1 + 2) * 3'
 ]
 
+{ #category : #'tests-send' }
+SlangBasicTranslationTest >> testSendTimesRepeat [
+
+	| translation send variable expression |
+	variable := TVariableNode new setName: 'var'.
+
+	expression := TSendNode new
+		              setSelector: #-
+		              receiver: variable
+		              arguments: { (TConstantNode value: 7) }.
+	send := TSendNode new
+		        setSelector: #timesRepeat:
+		        receiver: (TConstantNode value: 3)
+		        arguments: { (TStmtListNode new setStatements:
+				         { (TAssignmentNode new
+					          setVariable: variable
+					          expression: expression) }) }.
+	translation := self translate: send.
+
+	self assert: translation trimBoth equals: '{
+	var -= 7;
+}{
+	var -= 7;
+}{
+	var -= 7;
+}'
```

Comment:
This means that for small number of repeats the loop is unrolled? Or always? How many Times ? Could we add a comment?

Reply:
The loop is always unrolled, I will add a comment in the CAST method !
Btw in the slang method the comment is "Generate while(1) { stmtList } ", but that's not what the method is actually doing :P
I will change it too.